build_runner_core/lib/src/environment/create_merged_dir.dart

```diff
@@ -112,15 +113,12 @@ Future<bool> _createMergedOutputDir(
         await outputDir.create(recursive: true);
       }
 
-      outputAssets.addAll(await Future.wait(builtAssets.map((id) => _writeAsset(
-          id, outputDir, root, packageGraph, reader, symlinkOnly, hoist))));
-
-      var packagesFileContent = packageGraph.allPackages.keys
-          .map((p) => '$p:packages/$p/')
-          .join('\r\n');
-      var packagesAsset = AssetId(packageGraph.root.name, '.packages');
-      await _writeAsString(outputDir, packagesAsset, packagesFileContent);
-      outputAssets.add(packagesAsset);
+      outputAssets
+        ..addAll(await Future.wait(builtAssets.map((id) => _writeAsset(
+            id, outputDir, root, packageGraph, reader, symlinkOnly, hoist))))
+        ..add(await _writeCustomPackagesFile(packageGraph, outputDir))
+        ..add(await _writeModifiedPackageConfig(
+            packageGraph.root.name, reader, outputDir));
```

Comment:
[optional] Should we go for more in the `Future.wait`?

```suggestion
      outputAssets.addAll(await Future.wait([
          for(var id in builtAssets)
             _writeAsset(id, outputDir, root, packageGraph, reader, symlinkOnly, hoist),
           _writeCustomPackagesFile(packageGraph, outputDir),
           _writeModifiedPackageConfig(packageGraph.root.name, reader, outputDir),
       ]);
```